### ISSUE TITLE
"Run and Debug" and "F5" doesn't show the compilers detail in the API

### ISSUE THREAD

**Issue by ranolfi** [microsoft/vscode-cpptools#4278]
Type: General

- OS and Version: **Arch Linux**
- VS Code Version: **Code-OSS 1.37.1**
- C/C++ Extension Version: **0.25.1**

When we have a workspace folder open, in the Debug panel when clicking the Add Configuration option and then select _C++ (GDB/LLDB)_, some options with the same name are offered:

![image](https://user-images.githubusercontent.com/8507649/65284259-69f27300-db0f-11e9-9db6-8664905b02fa.png)

I tried all three "_g++ build and debug active file_". Options 1 and 2 generates the same `launch.json` and `tasks.json`; option 3 creates a slightly different `launch.json` file (`"miDebuggerPath": "/bin/gdb"` instead of `"/usr/bin/gdb"`).

In any case, this is confusing.

**Comment by elahehrashedi** [comment]
this issue is addressed in this PR: https://github.com/microsoft/vscode-cpptools/pull/5659

**Comment by elahehrashedi** [comment]
This is fixed in the new build task changes. Will be available in C/C++ Extension Version: 1.1.0

**Comment by sean-mcmanus** [comment]
@elahehrashedi I'm still reproing this with 1.1.0-insiders. What's supposed to happen?
![image](https://user-images.githubusercontent.com/19859882/95529692-7357b300-0990-11eb-9fa7-703da258c9fe.png)

**Comment by sean-mcmanus** [comment]
That screenshot is from F5 with no tasks.json generated.

**Comment by sean-mcmanus** [comment]
I see it fixed for Configure Default Build Task.

**Comment by elahehrashedi** [comment]
@sean-mcmanus can you please use the vscode-insiders so that I can see the details of each of those compilers?

**Comment by sean-mcmanus** [comment]
Yeah, it doesn't show the details with vscode-insiders. Use F5 (or Build and Debug Active File).

**Comment by elahehrashedi** [comment]
The user is getting this screen after running "Run and debug" or pressing "F5". I fixed this issue and sent a PR.
![image](https://user-images.githubusercontent.com/16143955/96952776-7bd9de80-14a4-11eb-89ed-58c38593ab91.png)
the paths are not visible:
![image](https://user-images.githubusercontent.com/16143955/96953165-4da8ce80-14a5-11eb-9c8f-1eac785fe0ea.png)
I solved this problem for "Run and Debug". look at bug: https://github.com/microsoft/vscode-cpptools/issues/6376

@sean-mcmanus but when we are running ("C/C++ build and debug active file), the details are not visible and it seems to be on Vs Code side, as the details are being set, but the API doesn't show them. I need to investigate more on this. If the details (i.e. compiler paths) are shown to the user, they will see the difference between different compilers. I opened a new issue for this.
